src/test/app/SetJSHook_test.cpp

```diff
@@ -2808,8 +2808,496 @@ class SetJSHook_test : public beast::unit_test::suite
         }
     }
 
-    // void
-    // test_emit(FeatureBitset features)
+    void
+    test_emit(FeatureBitset features)
+    {
+        testcase("Test emit");
+        using namespace jtx;
+        // Env env{*this, features};
+        Env env{
+            *this,
+            envconfig(),
+            features,
+            nullptr,  // beast::severities::kWarning
+            beast::severities::kTrace};
+
+        auto const alice = Account{"alice"};
+        auto const bob = Account{"bob"};
+        env.fund(XRP(10000), alice);
+        env.fund(XRP(10000), bob);
+
+        TestHook hook = jswasm[R"[test.hook](
+            const INVALID_ARGUMENT = -7
+            const PREREQUISITE_NOT_MET = -9
+            const EMISSION_FAILURE = -11
+            const sfDestination = ((8 << 16) + 3)
+            const ASSERT = (x) => {
+               if (!x) rollback(x.toString(), 0)
+            }
+            const Callback = (reserves) => {
+                // on callback we emit 2 more txns
+                const bob = otxn_field(sfDestination)
+                ASSERT(bob.length === 20)
+
+                ASSERT(otxn_generation() + 1 == etxn_generation())
+
+                ASSERT(etxn_burden() === PREREQUISITE_NOT_MET)
+
+                ASSERT(etxn_reserve(2) === 2)
+                
+                ASSERT(otxn_burden() > 0)
+                ASSERT(etxn_burden() === otxn_burden() * 2)
+
+                let tx = prepare({
+                    TransactionType: "Payment",
+                    Destination: util_raddr(bob),
+                    Amount: "1000"
+                })
+
+                const hash1 = emit(tx)
+                ASSERT(hash1.length === 32)
+
+                tx = prepare(tx)
```

Comment:
This results in the same hash, because you're passing it something with the sfEmitDetails already filled in, from the prior prepare

Reply:
This was inappropriate, so I will fix it.